### PR TITLE
Replace instances of ServiceFeatureTable.applyEditsAsync() with ServiceGeodatabase.applyEditsAysnc().

### PR DESCRIPTION
**Pull request by asantana917**

### CODE REVIEW

File: src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs [hunk 1]
```diff
@@ -36,13 +36,16 @@ public AddFeatures()
             Initialize();
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs:36]:
```suggestion
            _ = Initialize();
```
Make this change to resolve the warning.

File: src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs [hunk 2]
```diff
@@ -36,13 +36,16 @@ public AddFeatures()
...
         }
 
-        private void Initialize()
+        private async void Initialize()
         {
             // Create the map with streets basemap.
             MyMapView.Map = new Map(BasemapStyle.ArcGISStreets);
 
+            ServiceGeodatabase serviceGeodatabase = new ServiceGeodatabase(new Uri(FeatureServiceUrl));
+            await serviceGeodatabase.LoadAsync();
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs:45]: This await should be after the try. Maybe move the try higher in code.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditBranchVersioning/EditBranchVersioning.xaml.cs [hunk 3]
```diff
@@ -205,7 +205,7 @@ private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs
                     await _selectedFeature.FeatureTable.UpdateFeatureAsync(_selectedFeature);
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditBranchVersioning/EditBranchVersioning.xaml.cs:205]: On edit with branch versioning, the edited feature disappears after editing. It reappears when switching back and forth between versions.

**Review by ZackAllen — CHANGES_REQUESTED** [review]
The changes I mentioned for `Add features` on Xamarin.Forms apply for most of the changed samples. I ran everything on WPF, no behavior issues besides what I saw with branch versioning.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditBranchVersioning/EditBranchVersioning.xaml.cs (hunk 3 above)
Comment (asantana917) [reply src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/EditBranchVersioning/EditBranchVersioning.xaml.cs:205]: I tested the sample again be reverting the changes and this issue doesn't happen. I'm not sure if this sample is suppose to be changed with this issue but I'll reach out to Shubham and Mike to get more info.

**Review by nCastle1 — CHANGES_REQUESTED** [review]
The sample changes here need some additional comments in the code.

Additionally, I'd like to see the readmes for each of these samples updated. The current descriptions are no longer accurate with the updated code. Beyond just updating the steps, I'd like to see an explanation for why the updates are being applied to the underlying Geodatabase rather than the ServiceFeatureTable.

File: src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs [hunk 4]
```diff
@@ -36,36 +36,46 @@ public AddFeatures()
...
-            // Create the map with streets basemap.
-            MyMapView.Map = new Map(BasemapStyle.ArcGISStreets);
+            try
+            {
+                // Create the map with streets basemap.
+                MyMapView.Map = new Map(BasemapStyle.ArcGISStreets);
 
-            // Create the feature table, referring to the Damage Assessment feature service.
-            _damageFeatureTable = new ServiceFeatureTable(new Uri(FeatureServiceUrl));
+                ServiceGeodatabase serviceGeodatabase = new ServiceGeodatabase(new Uri(FeatureServiceUrl));
```
[the remark this answers is not in the document]
Comment (nCastle1) [reply src/Forms/Shared/Samples/Data/AddFeatures/AddFeatures.xaml.cs:47]: This is a good start, but I'd like to see more explanation, covering at least the following points:

- The feature table needs to be created from a `ServiceGeodatabase`. (I checked and the ServiceGeodatabase is null for tables created the usual way)
- The `ServiceGeodatabase` is needed in cases where there are rules/behaviors/relationships that span tables

File: src/Forms/Shared/Samples/Data/EditFeatureAttachments/EditFeatureAttachments.xaml.cs [hunk 5]
```diff
@@ -51,25 +51,32 @@ public EditFeatureAttachments()
...
         {
-            // Create the map with streets basemap.
-            MyMapView.Map = new Map(BasemapStyle.ArcGISStreets);
+            try
+            {
+                // Create the map with streets basemap.
+                MyMapView.Map = new Map(BasemapStyle.ArcGISStreets);
 
-            // Create the feature table, referring to the Damage Assessment feature service.
-            ServiceFeatureTable damageTable = new ServiceFeatureTable(new Uri(FeatureServiceUrl));
```
Comment (nCastle1) [inline src/Forms/Shared/Samples/Data/EditFeatureAttachments/EditFeatureAttachments.xaml.cs:60]: Why isn't this sample using the `ServiceGeodatabase`?

Comment (nCastle1) [inline src/Forms/Shared/Samples/Data/EditFeatureAttachments/EditFeatureAttachments.xaml.cs:60]: Why isn't this sample using the `ServiceGeodatabase`?

Comment (asantana917) [reply src/Forms/Shared/Samples/Data/EditFeatureAttachments/EditFeatureAttachments.xaml.cs:60]: @nCastle1 There is an issue with the data that is being used with this sample that is causing an error when using `ServiceGeodatabase`. The issue has been logged for [it](https://devtopia.esri.com/runtime/common-samples/issues/3379).

**Review by nCastle1 — APPROVED** [review]

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**